addons/knowledge/static/src/js/wysiwyg.js

```diff
@@ -0,0 +1,94 @@
+odoo.define('knowledge.wysiwyg', function (require) {
+'use strict';
+
+const core = require('web.core');
+const QWeb = core.qweb;
+
+const { DocumentWidget } = require('wysiwyg.widgets.media');
+const MediaDialog = require('wysiwyg.widgets.MediaDialog');
+const Link = require('wysiwyg.widgets.Link');
+const weWidgets = require('web_editor.widget');
+const Wysiwyg = require('web_editor.wysiwyg');
+
+const CustomDocumentWidget = DocumentWidget.extend({
+    /**
+     * @param {Object} img
+     * @returns {HTMLElement}
+     */
+    _renderMedia: function (img) {
+        let src = '';
+        if (img.image_src) {
+            src = img.image_src;
+            if (!img.public && img.access_token) {
+                src += _.str.sprintf('?access_token=%s', img.access_token);
+            }
+        }
+
+        const dom = $(QWeb.render('knowledge.file_block', {
+            img: img,
+            src: src
+        }));
+        this.$media = dom;
+        this.media = dom[0];
+
+        // Add mimetype for documents
+        if (!img.image_src) {
+            this.media.dataset.mimetype = img.mimetype;
+        }
+        this.$media.trigger('image_changed');
+        return this.media;
+    }
+});
+
+MediaDialog.include({
+    /**
+     * @param {Object} media
+     * @param {Object} options
+     * @returns
+     */
+    getDocumentWidget: function (media, options) {
```

Comment:
Looks like we are overriding the default behavior of the "MediaDialog", which calls a new Widget (without calling super).
That new widget itself overrides a whole method without calling super.

Seems dangerous and very low level, not sure what to think about it.